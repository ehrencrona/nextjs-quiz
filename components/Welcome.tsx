import theme from "../styles/theme";

const Welcome = ({ onStart }) => {
  return (
    <div>
      <h1>Welcome to the Trivia Challenge!</h1>

      <p>You will be presented with 10 true or false questions.</p>
      
      <p>Can you score 100%?</p>

      <button onClick={onStart}>Begin</button>

      <style jsx>{`
        h1 {
          font-size: 2em;
          font-weight: 600;
          letter-spacing: 0.01em;
          color: ${theme.colors.accent};
        }
      `}</style>
    </div>
  );
};

export default Welcome;
