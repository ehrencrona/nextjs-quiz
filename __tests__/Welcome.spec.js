import { fireEvent, render } from "@testing-library/react";
import React from "react";
import Welcome from "../components/Welcome";

describe("welcome page", () => {
  test("allows for starting", async () => {
    const onStart = jest.fn();

    const { getByText, container } = render(<Welcome onStart={onStart} />);

    expect(container).toMatchSnapshot();

    fireEvent.click(getByText("Begin"));

    expect(onStart).toHaveBeenCalled();
  });
});
