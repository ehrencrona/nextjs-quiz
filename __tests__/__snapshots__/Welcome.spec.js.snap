// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`welcome page allows for starting 1`] = `
<div>
  <div
    class="jsx-1356833245 page"
  >
    <div
      class="jsx-1356833245 content"
    >
      <h1
        class="jsx-2971654016"
      >
        Welcome to the Trivia Challenge!
      </h1>
      <p
        class="jsx-2971654016"
      >
        You will be presented with 10 true or false questions.
      </p>
      <p
        class="jsx-2971654016"
      >
        Can you score 100%?
      </p>
    </div>
    <div
      class="jsx-1356833245 buttons"
    >
      <div
        class="jsx-1356833245 fade"
      />
      <button
        class="jsx-1356833245"
      >
        Begin
      </button>
    </div>
  </div>
</div>
`;
